sympy/solvers/ode/single.py

```diff
@@ -784,5 +784,72 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [gensol]
 
 
+class SecondNonlinearAutonomousConserved(SinglePatternODESolver):
+    r"""
+    Gives solution for the autonomous second order nonlinear
+    differential equation of the form
+
+    .. math :: f''(x) = g(f(x))
+
+    The solution for this differential equation can be computed
+    by multiplying by `f'(x)` and integrating on both sides,
+    converting it into a first order differential equation.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, Symbol, dsolve
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+
+    >>> from sympy import sqrt
+    >>> eq = f(x).diff(x, 2) + sqrt(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(2*f(x)*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)*f(x)**(3/2)/(3*C1))/(3*sqrt(C1)*gamma(5/3)) - 2*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)/(3*C1))/(3*sqrt(C1)*gamma(5/3)), C2 + x),
+    Eq(2*f(x)*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)*f(x)**(3/2)/(3*C1))/(3*sqrt(C1)*gamma(5/3)) - 2*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)/(3*C1))/(3*sqrt(C1)*gamma(5/3)), C2 - x)]
+
+    >>> from sympy import exp, log
+    >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, 1, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, 1, f(x))), C2 - x)]
```

Comment:
A better example for the docstring would be the general case:
```python
In [3]: s1, s2 = dsolve(f(x).diff(x, 2) - g(f(x)), simplify=False)

In [4]: s1
Out[4]: 
f(x)                                   
 ⌠                                     
 ⎮             1                       
 ⎮   ────────────────────── du = C₂ + x
 ⎮       __________________            
 ⎮      ╱        ⌠                     
 ⎮     ╱  C₁ + 2⋅⎮ g(u) du             
 ⎮   ╲╱          ⌡                     
 ⌡                                     
 1                                     

In [5]: s2
Out[5]: 
f(x)                                   
 ⌠                                     
 ⎮             1                       
 ⎮   ────────────────────── du = C₂ - x
 ⎮       __________________            
 ⎮      ╱        ⌠                     
 ⎮     ╱  C₁ + 2⋅⎮ g(u) du             
 ⎮   ╲╱          ⌡                     
 ⌡                                     
 1    
```

Reply:
Okay, I'll replace the example that's too long with this.